models/components_test.go

```diff
@@ -85,15 +148,28 @@ var componentTestErrors = []componentTestError{
 	{"", ErrComponentFileDNE},
 	// Check loading a component with a broken schema
 	{filepath.Join("..", "fixtures", "component_fixtures", "EC2BrokenControl"), ErrControlSchema},
+	//Check loading an older schema without a key.
+	{filepath.Join("..", "fixtures", "component_fixtures", "EC2BadVersion2_0"), version.NewIncompatibleVersionError(version.NewRequirements(filepath.Join("..", "fixtures", "component_fixtures", "EC2BadVersion2_0", "component.yaml"), "component", semver.MustParse("2.0.0"), constants.MinComponentYAMLVersion, constants.MaxComponentYAMLVersion))},
+	//Check loading an older schema with a key.
+	{filepath.Join("..", "fixtures", "component_fixtures", "EC2WithKeyBadVersion2_0"), version.NewIncompatibleVersionError(version.NewRequirements(filepath.Join("..", "fixtures", "component_fixtures", "EC2WithKeyBadVersion2_0", "component.yaml"), "component", semver.MustParse("2.0.0"), constants.MinComponentYAMLVersion, constants.MaxComponentYAMLVersion))},
+	// Check for versions not in semver format.
+	{filepath.Join("..", "fixtures", "component_fixtures", "EC2BadVersion_NotSemver"), componentLoadError{fmt.Sprintf(constants.ErrVersionNotInSemverFormatf, 3)}},
```

Comment:
I don't think hard-coding the error strings is a terrible idea, to ensure that they are getting created as expected and there aren't any copy-and-paste errors.